contracts/AirdropCampaignData.sol

```diff
@@ -36,15 +40,22 @@ contract AirdropCampaignData {
         _;
     }
 
-    constructor(address _admin) {
+    // should this be onlyAdmin (backend wallet address),
+    // or should the deployer be able to upgrade
+    // (which means the cotract should have ownable)?
+    function _authorizeUpgrade(address) internal override onlyOwner {}
+
+    function initialize(address _admin) public initializer {
         admin = _admin;
+        __Ownable_init();
+        __UUPSUpgradeable_init();
     }
 
     /// @notice Updates the address of the admin variable
     /// @param _newAdmin - New address for the admin of this contract, and the address for all newly created airdrop contracts
     function changeAdmin(address _newAdmin) external onlyAdmin {
         admin = _newAdmin;
-        emit AdminChanged(_newAdmin);
```

Comment:
You don't really need this function + admin state variable + event anymore because `OwnableUpgradeable` is exposing `function transferOwnership(address newOwner) public`.

Reply:
Here is the problem with the naming convention I wanted to discuss...

On this contract, admin was meant to be the address of the wallet which can alter the state of this contract(submit data and so on).

Adding UUPSUpgradable, I had to change the name of the event `AdminChanged()` to `ContractAdminChanged()`, because the one contract included in UUPSUpgradable has an `AdminChanged()` event.

From what I can gather, this admin in UUPSUpgradable is the account which has permission to update the contract.

The variable name `admin` we initially defined does not clash with the `_admin` defined in UUPSUpgradable (unless I missed something). However, it might be an idea for us to change the name from admin to something else to better separate the two, since this may cause errors in the future.

What do you guys think?

Reply:
I'd rename it to something else to avoid confusion `AirbroManager, AirbroCampaignManager, Manager`
`admin` and `_admin` is really confusing

Reply:
I think that could work. `AirbroManager` seems to be the best fitting name. Will update.

Reply:
In fact, it should now be updated.